packages/typescript-estree/src/parseSettings/createParseSettings.ts

```diff
@@ -58,8 +64,14 @@ export function createParseSettings(
     programs: Array.isArray(options.programs) ? options.programs : null,
     projects: [],
     range: options.range === true,
-    singleRun: inferSingleRun(options),
+    singleRun,
     tokens: options.tokens === true ? [] : null,
+    tsconfigMatchCache: new ExpiringCache(
+      singleRun
+        ? 'Infinity'
+        : options.cacheLifetime?.glob ??
+          DEFAULT_TSCONFIG_CACHE_DURATION_SECONDS,
+    ),
```

Comment:
the parse settings are re-created every single time a new file is parsed - which means this cache will be discarded after a file is parsed.

it needs to be a module-level variable somewhere so we preserve it.